### ISSUE TITLE
create-nx-plugin throws ENOENT

### ISSUE THREAD

**Issue by GaryB432** [nrwl/nx#21684]
### Current Behavior

I am trying to create a new workspace to house many plugins each with many generators.  I think maybe I should just start with the integrated `npm` preset but see `create-nx-plugin` in the docs.  That command throws a ENOENT error on my windows computer.

### Expected Behavior

`create-nx-plugin` should work without errors.

### Steps to Reproduce

1. npx create-nx-plugin e2e-fun --ci=skip --directory=. --createPackageName=not-needed

### Nx Report

```shell
Node   : 20.10.0
   OS     : win32-x64
   npm    : 10.2.3

   nx             : 18.0.2
   @nx/js         : 18.0.2
   @nx/jest       : 18.0.2
   @nx/linter     : 18.0.2
   @nx/eslint     : 18.0.2
   @nx/workspace  : 18.0.2
   @nx/devkit     : 18.0.2
   @nx/plugin     : 18.0.2
   @nrwl/tao      : 18.0.2
   typescript     : 5.3.3
```

### Failure Logs

```shell
✔ Installing dependencies with npm
✔ Successfully created the workspace: e2e-fun.
node:events:492
      throw er; // Unhandled 'error' event
      ^

Error: spawn C:\WINDOWS\system32\cmd.exe ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:292:12)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn C:\\WINDOWS\\system32\\cmd.exe',
  path: 'C:\\WINDOWS\\system32\\cmd.exe',
  spawnargs: [
    '/d',
    '/s',
    '/c',
    '"npx nx --nxWorkspaceRoot="D:/g" g @nx/plugin:preset --quiet --interactive  --g --ci=skip --directory=. --createPackageName=not-needed --no-allPrompts --no-a --$0=C:\\Me\\_npx\\2f948f36b078f19f\\node_modules\\create-nx-plugin\\bin\\create-nx-plugin.js --name=e2e-fun --pluginName=e2e-fun --packageManager=npm"'
  ]
}

Node.js v20.10.0
```

### Package Manager Version

npm 10.2.3

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by AgentEnder** [comment]
Hey @GaryB432 👋 - sorry for the long response time on this one. I can't seem to repro the issue on windows. What terminal are you using?

**Comment by GaryB432** [comment]
Powershell 7 😁

**Comment by GaryB432** [comment]
@AgentEnder  bummer were you waiting for more information from me??

**Comment by AgentEnder** [comment]
Hey @GaryB432 👋,  sorry - need to get back to this one.

**Comment by GaryB432** [comment]
just reproduced the issue with my shell command as above on my MacBook fwiw. I guess I'm the only one at the `create-nx-plugin` party 😉. They don't know what they're missing.

**Comment by GaryB432** [comment]
update for august 2024.  still happening on 19.6.2

**Comment by GaryB432** [comment]
update for december 2024. still happening on 20.3.0

**Comment by laurentiu1981** [comment]
I had a similar issue when trying to run npx create-nx-workspace@latest
used powershell 7, with node 22.9.0 when I received a similar error.
Upgraded node to 22.16.0, problem went away.

**Comment by LayZeeDK** [comment]
> just reproduced the issue with my shell command as above on my MacBook fwiw. I guess I'm the only one at the `create-nx-plugin` party 😉. They don't know what they're missing.

What are the error details on macOS? Surely not `Error: spawn C:\WINDOWS\system32\cmd.exe ENOENT` 😅

I'm seeing this error on Windows 11 when I call `create-nx-workspace` using `exec`/`execSync` from `node:child_process` and pass a directory path on a different drive/file system than `C:\`, in my case a Windows Dev Drive (ReFS), `D:\`.

It might be because the `newGenerator` function calls `execSync` witout passing `{ env: process.env }` and/or `{ shell: true }`, see https://github.com/nrwl/nx/blob/60d019e0c72cfc8d6fc071af8cdb23af9056faef/packages/workspace/src/generators/new/new.ts#L79-L84 and https://stackoverflow.com/q/38458118/1071200.

It could be because the `ComSpec` environment variable is unavailable but `ENOENT` could also be thrown when an invalid `cwd` is passed to `exec`/`execSync`.

**Comment by GaryB432** [comment]
oh oopsie yes of course that was not the mac error and now I'ved backburnered my plugins and gone back to win11.  I will bust out the mbp and repro the error if necessary but yours sounds good!

**Comment by LayZeeDK** [comment]
I see in the issue description that you ran `create-nx-plugin` on your `D:\` drive, possibly also a Windows Dev Drive like me? I just verified that I get the same error as you, both on my `C:\` (NTFS) and `D:\` (Windows Dev Drive/ReFS) drives with the following command:

```powershell
# create-nx-plugin@latest == 22.5.0
npx create-nx-plugin@latest e2e-fun --ci=skip --directory=. --createPackageName=not-needed
```

However, when I remove `--directory=.`, no error occurs:

```powershell
# create-nx-plugin@latest == 22.5.0
npx create-nx-plugin@latest e2e-fun --ci=skip --createPackageName=not-needed
```

The same for not passing `.` to `--directory`, no errors:

```powershell
# create-nx-plugin@latest == 22.5.0
npx create-nx-plugin@latest e2e-fun --ci=skip --directory=e2e-fun --createPackageName=not-needed
```

Looking at [the Nx guide](https://nx.dev/docs/extending-nx/intro#create-your-own-plugin) and `npx create-nx-workspace --help`, I see no mention of `--directory` but it does seem to have an effect.

Where did you find the command you used with `--directory=.`?

**Comment by GaryB432** [comment]
I don't remember where specifically but I remember `directory` being introduced in many of the generators when `apps` and `libs` got seen as limiting I guess when project crystal surprised many of us.  I can confirm that, without `directory`, the repro instruction command creates what seems to be a legit plugin workspace. I notice that in terminal warnings it is referring to `@nx/plugin` as **third-party** which I'm sure is not as surprising to others who have been keeping up with nx in the intervening years.

```
'@nx/plugin' is not a built-in Nx preset.
Nx will download this npm package and run its preset generator.
Only proceed if you trust the publisher of '@nx/plugin'.
```

```
Node           : 24.18.0
OS             : linux-x64
Native Target  : x86_64-linux
npm            : 11.16.0
daemon         : Available

nx                 : 23.0.1
@nx/js             : 23.0.1
@nx/eslint         : 23.0.1
@nx/workspace      : 23.0.1
@nx/jest           : 23.0.1
@nx/devkit         : 23.0.1
@nx/eslint-plugin  : 23.0.1
@nx/plugin         : 23.0.1
typescript         : 5.9.3

```

**Comment by LayZeeDK** [comment]
Are you happy with this solution to your issue, @GaryB432?

**Comment by GaryB432** [comment]
Yes, thank you for the response. Please go ahead and close this issue. I will investigate this third party NX plugin and open another issue if I have one.
